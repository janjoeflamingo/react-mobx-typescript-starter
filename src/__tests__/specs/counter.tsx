// src/components/__tests__/counter_spec.tsx
import * as React from 'react'
// import * as TestUtils from 'react-dom/test-utils'
import * as ShallowRenderer from 'react-test-renderer/shallow'

// import { createStore } from 'redux'

import Counter from '../../pages/counter'
// import { reducers } from '../../reducers'

describe('<Counter />', () => {
  it('renders', () => {
    // const store = createStore(reducers)
    const renderer = ShallowRenderer.createRenderer()

    // <Counter store={store} />

    expect(renderer.render(<Counter initialCount={3} />)).toMatchSnapshot()
  })

  it('t-1', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<Counter initialCount={3} />)
    const result: any = renderer.getRenderOutput()
    expect(result.type.target).toBe('div')
  })
})
