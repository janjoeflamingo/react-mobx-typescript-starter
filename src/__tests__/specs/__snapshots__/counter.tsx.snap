// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders 1`] = `
<ForwardRef>
  <ForwardRef>
    3
  </ForwardRef>
  <ForwardRef
    onClick={[Function]}
  >
    Reset
  </ForwardRef>
  <ForwardRef
    onClick={[Function]}
  >
    +
  </ForwardRef>
  <ForwardRef
    onClick={[Function]}
  >
    -
  </ForwardRef>
</ForwardRef>
`;
